include/alpaka/dev/DevGenericSycl.hpp

```diff
@@ -112,11 +113,41 @@ namespace alpaka
                 return m_context;
             }
 
+            auto deviceProperties() -> std::optional<alpaka::DeviceProperties>&
+            {
+                std::call_once(
+                    m_onceFlag,
+                    [&]()
+                    {
+                        m_deviceProperties = std::make_optional<alpaka::DeviceProperties>();
+                        auto const device = this->get_device();
```

Comment:
can this be a `const&`
```suggestion
                        auto const& device = this->get_device();
```
?